console/src/components/Services/Common/ComponentErrorBoundary.tsx

```diff
@@ -0,0 +1,68 @@
+import React, { Component, ReactNode } from 'react';
+import globals from '../../../Globals';
+import styles from '../../Error/ErrorPage.scss';
+
+const errorImageSrc = `${globals.assetsPath}/common/img/hasura_icon_green.svg`;
+
+interface ErrorState {
+  error: Error;
+  hasError: boolean;
+}
+
+type Props = {
+  children: ReactNode;
```

Comment:
You don't need to specify this in `Props`. It will be included as your component extends `React.Component`.

Btw. maybe we can convert it to functional component?

Reply:
we need it to be a class component for the `componentDidCatch` lifecycle method.